packages/components/src/components/hds/tabs/tab.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { guidFor } from '@ember/object/internals';
+import { action } from '@ember/object';
+import type { IconName } from '@hashicorp/flight-icons/svg';
+import type { HdsTabsTabIds } from './types';
+
+export interface HdsTabsTabSignature {
+  Args: {
+    selectedTabIndex: number;
+    tabIds: HdsTabsTabIds;
+    count?: string;
+    icon?: IconName;
+    isSelected?: boolean;
+    didInsertNode: (element: HTMLButtonElement, isSelected?: boolean) => void;
+    didUpdateNode: (nodeIndex: number, isSelected?: boolean) => void;
+    willDestroyNode: (element: HTMLButtonElement) => void;
+    onClick: (event: MouseEvent, tabIndex: number) => void;
+    onKeyUp: (nodeIndex: number, event: KeyboardEvent) => void;
+  };
+  Blocks: {
+    default: [];
+  };
+  Element: HTMLLIElement;
+}
+
+export default class HdsTabsTabComponent extends Component<HdsTabsTabSignature> {
+  /**
+   * Generate a unique ID for the Tab
+   * @return {string}
+   */
+  tabId = 'tab-' + guidFor(this);
+
+  get nodeIndex(): number | undefined {
+    return this.args.tabIds?.indexOf(this.tabId);
+  }
+
+  /**
+   * Determine if the tab is the selected tab
+   * @return {boolean}
+   * @default false (1st tab is selected by default)
+   */
+  get isSelected(): boolean {
+    return (
+      this.nodeIndex !== undefined &&
+      this.nodeIndex === this.args.selectedTabIndex
+    );
+  }
+
+  @action
+  didInsertNode(element: HTMLButtonElement, positional: [boolean?]) {
+    const { didInsertNode } = this.args;
+
+    const isSelected = positional[0];
+
+    if (typeof didInsertNode === 'function') {
+      didInsertNode(element, isSelected);
+    }
+  }
+
+  @action
+  didUpdateNode() {
+    const { nodeIndex } = this;
+    const { didUpdateNode } = this.args;
+
+    if (typeof didUpdateNode === 'function' && nodeIndex !== undefined) {
+      didUpdateNode(nodeIndex, this.args.isSelected);
+    }
+  }
+
+  @action
+  willDestroyNode(element: HTMLButtonElement) {
+    const { willDestroyNode } = this.args;
+
+    if (typeof willDestroyNode === 'function') {
+      willDestroyNode(element);
+    }
+  }
+
+  @action
+  onClick(event: MouseEvent) {
+    const { nodeIndex } = this;
```

Comment:
in the original `.js` file this was not destructured and we simply used `this.nodeIndex` in the code below. I have tried to revert this change and run `yarn lint:types` and the tests passed. 🤔

Reply:
Destructuring made sense to me because we're using the value from `this.nodeIndex` in more than one place now.

Reply:
Is this ok or something you think we should revert before merging?

Reply:
not a blocker